skimage/measure/entropy.py

```diff
@@ -0,0 +1,33 @@
+from scipy.stats import entropy as scipy_entropy
+from .._shared.utils import assert_nD
+
+
+def shannon_entropy(image):
+    """
+    Calculate the Shannon entropy of a grayscale image.
+
+    The Shannon entropy is defined as S = -sum(pk * log(pk)),
+    where pk are the number of pixels of value k.
+
+    Parameters
+    ----------
+    image : (N, M) ndarray
+        Grayscale input image.
+
+    Returns
+    -------
+    entropy : float
+
+    References
+    ----------
+    .. [1] https://en.wikipedia.org/wiki/Entropy_(information_theory)
+    .. [2] https://en.wiktionary.org/wiki/Shannon_entropy
+
+    Examples
+    --------
+    >>> from skimage import data
+    >>> shannon_entropy(data.camera())
+    12.290907503512717
+    """
+    assert_nD(image, 2)
+    return scipy_entropy(image.ravel())
```

Comment:
I have a slight preference towards using base 2... Do you have a preference for base e?